plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetastoreAccessOperations.java

```diff
@@ -123,7 +123,7 @@ public void testSelect()
 
         assertMetastoreInvocations("SELECT * FROM test_select_from",
                 ImmutableMultiset.builder()
-                        .add(GET_TABLE)
+                        .addCopies(GET_TABLE, 2)
```

Comment:
Why does "Improve Iceberg commit to HMS" change the invocation counts on the read path?

Reply:
The first one is `TrinoHiveCatalog` getting the table to check if the table is a view/materialized view. The second call is `AbstractMetastoreTableOperations` loading the Iceberg table, when it does that the cache is invalidated to get the most recent Snapshot.

I couldn't find a good way to invalidate the cache only during commit retries and not during initial table loading, they use the same code path. `getRefreshedLocation`

Reply:
Updated to include an `invalidateCaches` parameter, which is set to true when `refresh` is called, and false when `current` is called. That meets the expectations of the Iceberg library